samples/distributedtracing/distributedtracing.go

```diff
@@ -93,7 +137,13 @@ func DoWorkActivity(ctx task.ActivityContext) (any, error) {
 	}
 
 	// Simulate doing work
-	time.Sleep(duration)
+	select {
+	case <-time.After(duration):
+		// Ok
+	case <-ctx.Context().Done():
```

Comment:
Is the reason to make this change so that he sleep can be cancelled?

Reply:
Yes, it's just a pattern. If there's a context passed to the method, callers expect to be able to cancel the execution of the method (at least, while it's in a waiting state) by canceling the context.

Since this is a sample, it seemed a good place to showcase a best practice